syncstorage/tests/test_sql.py

```diff
@@ -197,3 +198,21 @@ def count_items():
         self.assertEquals(count_items(), 5)
         self.assertEquals(len(self.storage.get_items(_USER, "col")["items"]),
                           5)
+
+    def test_migrating(self):
+        migrating_id = str(uuid.uuid4())
```

Comment:
hrm, this is a bit tricky. The best way to test this is to add a test under `tests/functional/test_storage.py` which would add a record to a migration table and then test that the middleware layer returns a 503 when a matching fxa_uid record request is made. The problem is that there's a significant abstraction layer between the test architecture and the wsgi app such that I can't get to the wsgi storage from inside of the test framework. Since it's using `sqlite:////memory` that also complicates things. 

Another problem is that the `fxa_uid` is set to a random UUID in the `_authenticate()` function in `tests/function/support` so I can't just add a `testing` fxa_uid easily. 

I'm not really inclined to add a "set this user as migrating" type of call, even for testing, because that could be abused should it ever hit production. 

Any other test basically boils down to what's this test is doing, since the decorator is attached to all the relevant endpoints. 

I'll see if there's a way to get at least one full test, though.

Reply:
Nevermind, I'm dumb and didn't read the patch below.

Reply:
lol; in this case  meant something a bit simpler though, basically taking the "assert that `is_migrating` ends up returning True" check from the bottom of this test, and copying it up here to the top as an "assert that `is_migrating` started out returning False" check.